api/v1beta2/mysqlcluster_types.go

```diff
@@ -259,17 +266,96 @@ func (s MySQLClusterSpec) validateUpdate(old MySQLClusterSpec) field.ErrorList {
 			newSize := pvc.StorageSize()
 			oldSize := old.StorageSize()
 
-			if newSize.Cmp(oldSize) == -1 {
+			switch cmp := newSize.Cmp(oldSize); {
+			case cmp == -1:
 				p := p.Child("volumeClaimTemplates").Index(i).
 					Child("spec").Child("resources").Child("requests").Key("storage")
 				allErrs = append(allErrs, field.Forbidden(p, "storage size cannot be reduced"))
+			case cmp == 1:
+				isVolumeExpansion = true
+			case cmp == 0:
+				// noop
 			}
 		}
 	}
 
+	if isVolumeExpansion {
+		allErrs = append(allErrs, s.validateVolumeExpansionSupported(ctx, apiReader)...)
+	}
+
 	return append(allErrs, s.validateCreate()...)
 }
 
+func (s MySQLClusterSpec) validateVolumeExpansionSupported(ctx context.Context, apiReader client.Reader) field.ErrorList {
+	var allErrs field.ErrorList
+	p := field.NewPath("spec").Child("volumeClaimTemplates")
+
+	for i, pvc := range s.VolumeClaimTemplates {
+		if pvc.Spec.StorageClassName != nil {
+			var sc storagev1.StorageClass
+			if err := apiReader.Get(ctx, types.NamespacedName{Name: *pvc.Spec.StorageClassName}, &sc); err != nil {
+				p := p.Index(i).Child("spec").Child("storageClassName")
+				allErrs = append(allErrs, field.InternalError(p, fmt.Errorf("failed to get storage class %s: %w", *pvc.Spec.StorageClassName, err)))
+			} else {
+				if !isVolumeExpansionSupported(sc) {
+					p := p.Index(i).Child("spec").Child("storageClassName")
+					allErrs = append(allErrs, field.Forbidden(p, fmt.Sprintf("storage class %q is not allowed to expand volume", *pvc.Spec.StorageClassName)))
+				}
+			}
+		} else {
+			sc, err := getDefaultStorageClass(ctx, apiReader)
+			if err != nil {
+				p := p.Index(i)
+				allErrs = append(allErrs, field.InternalError(p, fmt.Errorf("failed to get storage class: %w", err)))
+			} else {
+				if !isVolumeExpansionSupported(sc) {
+					p := p.Index(i)
+					allErrs = append(allErrs, field.Forbidden(p, fmt.Sprintf("default storage class %q is not allowed to expand volume", sc.Name)))
+				}
+			}
+		}
+	}
+
+	return allErrs
+}
+
+func isVolumeExpansionSupported(sc storagev1.StorageClass) bool {
```

Comment:
Could you pass the pointer? 

```suggestion
func isVolumeExpansionSupported(sc *storagev1.StorageClass) bool {
```

Reply:
Thanks, I fixed:
https://github.com/cybozu-go/moco/pull/420/commits/f70217e23e35abab3a3fe5e7ed6970a69dd89b4e